src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClientProvider.cs

```diff
@@ -27,20 +27,30 @@ public Factory()
             {
             }
 
+            [Obsolete(MefConstruction.FactoryMethodMessage, error: true)]
             public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
-                => new ServiceHubRemoteHostClientProvider(workspaceServices);
+            {
+                if (!RemoteHostOptions.IsServiceHubProcess64Bit(workspaceServices)
```

Comment:
>!RemoteHostOptions.IsServiceHubProcess64Bit(workspaceServices) [](start = 20, length = 62)

This is confusing. If IsServiceHubProcess64Bit returns false we will run in-proc, not 32-bit ServiceHub.

Reply:
I will split it into two separate options for clarity.

Reply:
➡️ Added `RemoteHostOptions.IsUsingServiceHubOutOfProcess` for clarity.